defined.ts

```diff
@@ -0,0 +1,9 @@
+const defined =
+  <T>(value: T): Exclude<T, undefined> => {
```

Comment:
I don't think that this excludes any undefined from the given type.

From ```
type t = {
  type: string;
  arg: undefined;
}
``` 

if we do `Exclude<t, undefined>` the same type is returned

Reply:
Yes it does https://www.typescriptlang.org/play?#code/C4TwDgpgBAglC8UDOwBOBLAdgcygHygFdMATCAMywhIChRIoAhBKAUQA8BjAG0LIB4YAGiKkKVEgD4aNIA